pkg/metrics/metrics.jsx

```diff
@@ -893,21 +896,31 @@ class MetricsHistory extends React.Component {
         this.handleSelect = this.handleSelect.bind(this);
         this.handleInstall = this.handleInstall.bind(this);
 
-        // load and render the last 24 hours (plus current one) initially; this needs numCpu initialized for correct scaling
-        // FIXME: load less up-front, load more when scrolling
-        machine_info_promise.then(() => {
-            cockpit.spawn(["date", "+%s"])
-                    .then(out => {
-                        const now = parseInt(out.trim()) * 1000;
-                        const current_hour = Math.floor(now / MSEC_PER_H) * MSEC_PER_H;
-                        this.load_data(current_hour - LOAD_HOURS * MSEC_PER_H, undefined, true);
-                        this.today_midnight = new Date(current_hour).setHours(0, 0, 0, 0);
-                        this.setState({
-                            selectedDate: this.today_midnight,
-                        });
-                    })
-                    .catch(ex => this.setState({ error: ex.toString() }));
+        /* supervise pmlogger.service, to diagnose missing history */
+        this.pmlogger_service = service.proxy("pmlogger.service");
+        this.pmlogger_service.addEventListener("changed", () => {
+            if (this.pmlogger_service.state !== this.state.pmLoggerState)
```

Comment:
Is this check necessary?

Reply:
Not strictly -- just because this page is rather expensive to render, I thought I'd rather go through some extra effort to not call setState() unnecessarily.

Reply:
I mean, even calling render() itself is expensive (not the actual screen update, as shadow DOM would of course optimize that away)

Reply:
This is also the reason why I'm not using our fancy new `useEvent(proxy, "changed")`. There are about 4 successive events with mostly redundant data (for the particular `state` property that we are interested in).

Reply:
right, makes sense and cannot hurt :)